### PR TITLE
TIP-733: Separate grid toolbar modules into form extensions

### PR DESCRIPTION
**Pull request by tamarasaurus**
**Description (for Contributor and Core Developer)**

This PR separates the components of the grid toolbar into form extensions, including:

- [x] Pagination
- [x] Bulk actions
- [x] Quick export
- [x] Column selection

It removes
- [x] View per page (now defaults to 100)
- [x] Reset
- [x] Refresh

See https://github.com/akeneo/pim-community-dev/pull/6524 for full description 

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/action/configure-columns-action.js [hunk 1]
```diff
@@ -64,35 +70,35 @@ define(
...
             ),
 
-            initialize: function (options) {
+            initialize: function () {
+                mediator.once('grid_load:start', this.setupOptions.bind(this));
+
+                 BaseForm.prototype.initialize.apply(this, arguments);
+            },
+
+            setupOptions: function(collection, gridContainer) {
```
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/action/configure-columns-action.js]: jsdoc?

Comment (tamarasaurus) [reply src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/action/configure-columns-action.js]: fixed :+1:

File: features/mass-action/select_products_to_mass_edit.feature [hunk 2]
```diff
@@ -43,13 +43,11 @@ Feature: When I mass edit I should be able to see how many items will be edited
     Then I should see the text "Select your action for the 19 products"
 
   Scenario: Successfully count the number of mass-edited items when click on all visible products
-    Given I change the page size to 10
     And I select all visible entities
     When I press "Change product information" on the "Bulk Actions" dropdown button
-    Then I should see the text "Select your action for the 10 products"
+    Then I should see the text "Select your action for the 19 products"
```
Comment (pierallard) [inline features/mass-action/select_products_to_mass_edit.feature]: You can drop this test (or create 101 products)

File: features/reference-data/product/filtering/filtering_by_multi_reference_data.feature [hunk 3]
```diff
@@ -21,9 +21,10 @@ Feature: Filter products by reference data
     And I am logged in as "Mary"
     And I am on the products grid
 
+  @skip @info To be unskipped in PIM-6574
   Scenario: Successfully filter product with multi reference data filters
-    # Given I show the filter "sole_color"
-    # And I filter by "sole_color" with operator "in list" and value "Red"
+    Given I show the filter "sole_color"
+    And I filter by "sole_color" with operator "in list" and value "Red"
```
Comment (pierallard) [inline features/reference-data/product/filtering/filtering_by_multi_reference_data.feature]: No interest to have these 2 lines, drop them

**Pull request merged**